src/triage/component/catwalk/baselines/thresholders.py

```diff
@@ -0,0 +1,184 @@
+import numpy as np
+import pandas as pd
+from six import string_types
+
+from triage.component.catwalk.exceptions import BaselineFeatureNotInMatrix
+
+REQUIRED_KEYS = ['feature_name', 'operator', 'threshold']
+
+
+class SimpleThresholder(object):
+    """ The simple thresholder applies a set of predetermined logical rules to a
+    test matrix to classify entities. By default, it will classify entities as 1
+    if they satisfy any of the rules. When 'or' is set as the logical_operator
+    attribute, it will classify entities as 1 if they pass *all* of the rules.
+
+    Rules are passed as either strings in the format 'x1 > 5' or dictionaries in
+    the format {feature_name: 'x1', operator: '>', threshold: 5}. The
+    feature_name, operator, and threshold keys are required. Eventually, this
+    class may be abstracted into a BaseThreshold class and more complicated
+    thresholders could be built around new keys in the dictionaries (e.g., by
+    specifying scores that could be applied (and possibly summed) to entities
+    satisfying rules) or by an alternative dictionary format that specifies
+    more complicated structures for applying rules (for example:
+        {
+            or: [
+                {or: [{}, {}]},
+                {and: [{}, {}]}
+            ]
+        }
+    where rules and operators that combine them can be nested).
+    """
+    def __init__(self, rules, logical_operator='or'):
+        self.rules = rules
+        self.logical_operator = logical_operator
+        self.feature_importances_ = None
+        self.rule_combination_method = self.lookup_rule_combination_method(logical_operator)
+
+    @property
+    def rules(self):
+        return vars(self)['rules']
+
+    @rules.setter
+    def rules(self, rules):
+        """ Validates the rules passed by the user and converts them to the
+        internal representation. Can be used to validate rules before running an
+        experiment.
+
+        1. If rules are not a list, make them a list.
+        2. If rules are strings, convert them to dictionaries.
+        3. If dictionaries or strings are not in a supported format, raise
+           helpful exceptions.
+        """
+        if not isinstance(rules, list):
+            rules = [rules]
+
+        converted_rules = []
+        for rule in rules:
+            if isinstance(rule, string_types):
+                converted_rules.append(self._convert_string_rule_to_dict(rule))
+            else:
+                if not isinstance(rule, dict):
+                    raise ValueError((
+                        'Rule "{rule}" is not of a supported type (string or '
+                        'dict).'.format(rule=rule)
+                    ))
+                if not rule.keys() >= REQUIRED_KEYS:
```

Comment:
I think this expression will raise `TypeError`, if `REQUIRED_KEYS` is `list`. Rather, you'll want some sort of `set`.

Since it's a module-constant, just wrap it in `frozenset`:

    REQUIRED_KEYS = frozenset(['a', 'b', 'c'])